### PR TITLE
UG-646 Handle config within RPCO [Backport: newton-14.0]

### PR DESCRIPTION
**Pull request by hughsaunders**
Currently the rpc-gating scripts drop some RPCO config overrides,
some are always added, others are scenario dependant. This is
problematic as those overrides may not work for every branch of RPCO
and it creates a tight and inflexible coupling between RPCO and gating.

This commit adds the overrides that gating currently drops into
the boostrap-aio.yml playbook. These overrides are conditionally
applied based on environment variables that indicate the type of
build that is required.

This duplicates gating functionality so must not be merged until it's
counterpart in rpc-gating is merged:
  https://github.com/rcbops/rpc-gating/pull/266

(cherry picked from commit 4435dda995de74894017ba48eeced19a3ebe4715)
(cherry picked from commit 72de4174739131c07aea98dedaae320e0b091023)

Issue: [UG-646](https://rpc-openstack.atlassian.net/browse/UG-646)

### CODE REVIEW

**Comment by hughsaunders** [comment]
recheck cit_trusty_deploy_swift failed on test server basic ops

**Comment by hughsaunders** [comment]
recheck_cit_trusty_deploy_swift failed on test server basic ops

File: ansible-role-requirements.yml [hunk 1]
```diff
@@ -14,3 +14,7 @@
   scm: git
   src: https://github.com/rsoprivatecloud/openstack-ops
   version: 1383d0db302c6fc400a50fae4f85e77bf1633581
```
Comment (mattt416) [inline ansible-role-requirements.yml]: Can we add a note about this needing to come out, as exists here:

https://github.com/rcbops/rpc-openstack/blob/newton-14.1/ansible-role-requirements.yml#L21-L23

Comment (hughsaunders) [inline ansible-role-requirements.yml]: Done.

**Pull request merged**